ui/src/pages/Recovery/Recover.js

```diff
@@ -0,0 +1,227 @@
+import { useEffect, useState } from "react"
+import { appWindow, LogicalSize } from '@tauri-apps/api/window';
+import { Outlet } from "react-router-dom";
+import { useNavigate, useLocation } from "react-router-dom";
+const bip39 = require('bip39');
+
+const MIN_PASSWORD_LENGTH = 8;
+
+const DEFAULT_WINDOW_SIZE = new LogicalSize(460, 500);
+const CONFIRM_PHRASE_WINDOW_SIZE = new LogicalSize(460, 750);
+
+const DROPDOWN_OPTIONS = [
+  {
+    text: "I have a 12 word phrase",
+    value: 12
+  },
+  {
+    text: "I have a 18 word phrase",
+    value: 18
+  },
+  {
+    text: "I have a 24 word phrase",
+    value: 24
+  },
+];
+
+const DEFAULT_PHRASES = {
+  12: ["", "", "", "", "", "", "", "", "", "", "", ""],
+  18: ["", "", "", "", "", "", "", "", "", "", "", "",
+    "", "", "", "", "", ""],
+  24: ["", "", "", "", "", "", "", "", "", "", "", "",
+    "", "", "", "", "", "", "", "", "", "", "", ""]
+}
+
+// By default this component will load using a 12 word phrase.
+const Recover = ({
+  payloadType,
+  sendSelection,
+  restartServer,
+  sendPassword,
+  sendMnemonic
+}) => {
+
+  // recovery phrase
+  const [mnemonics, setMnemonics] = useState(DEFAULT_PHRASES[12]);
+  const [mnemonicsValidity, setMnemonicsValidity] = useState(false);
+  const [validMnemonics, setValidMnemonics] = useState(null);
+
+  // new passwords
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [passwordsMatch, setPasswordsMatch] = useState('');
+  const [isValidPassword, setIsValidPassword] = useState(false);
+  const [showError, setShowError] = useState(false);
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  useEffect(() => {
+
+    const checkPasswordMatch = () => {
+      if (password === confirmPassword) {
+        setPasswordsMatch(true);
+      } else {
+        setPasswordsMatch(false);
+      }
+    }
+
+    const checkPasswordValidity = () => {
+      if (isValid(password)) {
+        setIsValidPassword(true);
+      } else {
+        setIsValidPassword(false);
+      }
+    }
+
+    checkPasswordMatch();
+    checkPasswordValidity();
+
+  }, [password, confirmPassword, isValidPassword, passwordsMatch]);
```

Comment:
as elsewhere, you don't want `isValidPassword ` or `passwordsMatch` in the dependencies array here